packages/core/core/src/configuration/urls.ts

```diff
@@ -43,20 +43,16 @@ export const getConfigUrls = (config: Record<string, unknown>, forAdminBuild = f
       throw new Error('Invalid admin url config. Make sure the url defined in server.js is valid.');
     }
   } else {
+    // in case when adminUrl is not an absolute url, we need to prepend serverUrl
+    // to make it relative to the server, ie /admin or /strapi/admin (if serverUrl is /strapi)
     adminUrl = `${serverUrl}/${adminUrl}`;
   }
 
   // Defines adminPath value
+  // adminPath is the path of the admin panel relative to the host
+  // this could be just /admin, or custom one like /dashboard, or subpath like /strapi/dashboard
   let adminPath = adminUrl;
-  if (
-    serverUrl.startsWith('http') &&
-    adminUrl.startsWith('http') &&
-    new URL(adminUrl).origin === new URL(serverUrl).origin &&
-    !forAdminBuild
-  ) {
-    adminPath = adminUrl.replace(strings.getCommonPath(serverUrl, adminUrl), '');
-    adminPath = `/${_.trim(adminPath, '/')}`;
```

Comment:
What's the main reason behind removing this bit?

Reply:
@Convly thanks for this question! this is related to the issue of loading admin panel's resources, when we setup the configuration is such way that admin panel is in subpath to the strapi instance (ie. `https://domain.fake/strapi` and `https://domain.fake/strapi/admin`) the code above will remove the `https://domain.fake/strapi` from the `adminPath`. Later in the code, this variable is used to set the `ctx.basePath` which is used to put the links to the resources in admin panel causing the issue loading it because it will be looking for resources at `https://domain.fake` instead of `https://domain.fake/strapi`